piquasso/api/branch.py

```diff
@@ -0,0 +1,102 @@
+#
+# Copyright 2021-2025 Budapest Quantum Computing Group
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Tuple, Union
+
+from .state import State
+
+from fractions import Fraction
+
+
+class Branch:
+    r"""Represents a branching in the computation caused by a measurement.
+
+    Describes a branch in the computation tree that arises due to a measurement
+    operation. Each branch contains the post-measurement quantum state, the outcomes
+    of the measurements that led to this branch, and the frequency (or weight) of
+    this branch in the overall ensemble. The frequency indicates how often this
+    particular outcome occurred relative to the total number of shots (simulations)
+    performed.
```

Comment:
Two things could elevate this explanation:
1. What an ensemble is in this context (I'm not sure all readers would be familiar with it)
2. The connection between frequency and probability.